toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/ServerTrustAuthenticator.kt

```diff
@@ -18,49 +18,193 @@
 
 package com.arcgismaps.toolkit.authentication
 
-import androidx.compose.material3.AlertDialog
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
 import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextButton
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.arcgismaps.httpcore.authentication.NetworkAuthenticationChallenge
+import com.arcgismaps.httpcore.authentication.NetworkAuthenticationType
+
+/**
+ * Displays a trust or distrust server prompt to the user in a Dialog.
+ *
+ * @param serverTrustChallenge the pending [ServerTrustChallenge] that initiated this prompt.
+ * @param modifier the [Modifier] to be applied to this ServerTrustAuthenticator.
+ * @since 200.6.0
+ */
+@Composable
+internal fun ServerTrustAuthenticatorDialog(
+    serverTrustChallenge: ServerTrustChallenge,
+    modifier: Modifier = Modifier
+) {
+    Dialog(
+        onDismissRequest = {
+            serverTrustChallenge.distrust()
+        }
+    ) {
+        Card(
+            colors = CardDefaults.cardColors(
+                containerColor = MaterialTheme.colorScheme.surface,
+            ),
+            shape = RoundedCornerShape(16.dp)
+        ) {
+            ServerTrustAuthenticatorImpl(
+                serverTrustChallenge.challenge.hostname,
+                modifier,
+                onConfirm = { serverTrustChallenge.trust() },
+                onCancel = { serverTrustChallenge.distrust() }
+            )
+        }
+    }
+}
 
 /**
  * Displays a trust or distrust server prompt to the user.
  *
  * @param serverTrustChallenge the pending [ServerTrustChallenge] that initiated this prompt.
  * @param modifier the [Modifier] to be applied to this ServerTrustAuthenticator.
- * @since 200.2.0
+ * @since 200.6.0
```

Comment:
I think we should keep this as `200.2.0` just to not confuse ourselves.
```suggestion
 * @since 200.2.0
```